### ISSUE TITLE
Wrong counting of roundabout exits?

### ISSUE THREAD

**Issue by gunnar22** [osmandapp/OsmAnd#9641]
I used OsmAnd with navigation for car. At a typical 4-way roundabout. I wanted to go straight ahead, i.e., take the second exit. OsmAnd told me to take the first exit. 
OSM tags on first exit: 
- highway=unclassified
- motor_vehicle=private

Theory: Does OsmAnd think that because it is not legal for me to take the first exit, OsmAnd should not count the first exit, and instead enumerate the second exit as the first exit, and say "take the first exit".

It would be better if OsmAnd counted and enumerated all exits, to make it quick to figure out visually which exit to take. 

https://www.openstreetmap.org/#map=16/63.3221/10.3788
Driving north-west on Brøttemsvegen. First exit is Tilleråsen.

(Using OsmAnd+ 3.7.4, offline maps, offline navigation.)

**Comment by vshcherb** [comment]
Yes you are absolutely right. The problem is how we structure the map in OSM, for example there are 3 exits though in OSM there are typically 6 oneway roads connected to roundabout because 3 are forbidden, so we can count correctly. 

It is not only oneway, it is also important to take into account access=no+bicycle=yes etc.

Private is a special cause they shouldn't be forbidden and they are not forbidden if you click a button allow private roads. Though our algorithm is quite generic today, so we need to distinguish somehow not-forbidden to count but forbidden to drive

**Comment by Kjarrval** [comment]
Perhaps this is related to a routing error because recently I've started to receive instructions to constantly take the first exit in roundabouts when it was usually and really the second or even third exit, when it hadn't happened in that particular route before.

**Comment by vshcherb** [comment]
https://www.openstreetmap.org/way/837265490
